Client/Frontend/AuthenticationManager/PasscodeEntryViewController.swift

```diff
@@ -7,18 +7,13 @@ import SnapKit
 import Shared
 import SwiftKeychainWrapper
 
-/// Delegate available for PasscodeEntryViewController consumers to be notified of the validation of a passcode.
-@objc protocol PasscodeEntryDelegate: AnyObject {
-    func passcodeValidationDidSucceed()
-    @objc optional func userDidCancelValidation()
-}
-
 /// Presented to the to user when asking for their passcode to validate entry into a part of the app.
 class PasscodeEntryViewController: BasePasscodeViewController {
-    weak var delegate: PasscodeEntryDelegate?
+    fileprivate let passcodeCompletion: ((Bool) -> Void)
```

Comment:
Could have been a deferred, but it seemed simpler in this case as a completion block.